src/app/Http/Controllers/Operations/InlineCreateOperation.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Backpack\CRUD\app\Http\Controllers\Operations;
+
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+use Illuminate\Support\Facades\Route;
+
+trait InlineCreateOperation
+{
+    /**
+     * Define which routes are needed for this operation.
+     *
+     * @param string $segment    Name of the current entity (singular). Used as first URL segment.
+     * @param string $routeName  Prefix of the route name.
+     * @param string $controller Name of the current CrudController.
+     */
+    protected function setupInlineCreateOperationRoutes($segment, $routeName, $controller)
+    {
+        Route::get($segment.'/inline/create', [
+            'as'        => $segment.'-inline-create',
+            'uses'      => $controller.'@getInlineCreateModal',
+            'operation' => 'InlineCreateOperation',
+        ]);
+        Route::post($segment.'/inline/create', [
+            'as'        => $segment.'-inline-create-save',
+            'uses'      => $controller.'@storeInlineCreate',
+            'operation' => 'InlineCreateOperation',
+        ]);
+        Route::get($segment.'/inline/refresh', [
+            'as'        => $segment.'-inline-refresh-options',
+            'uses'      => $controller.'@inlineRefreshOptions',
+            'operation' => 'InlineCreateOperation',
+        ]);
+    }
+
+    public function setupInlineCreateDefaults()
+    {
+        $this->crud->setOperationSetting('inline_create', true);
+    }
+
+    public function getInlineCreateModal()
+    {
+        if (request()->has('entity')) {
+            $this->setupOperationSettings();
+
+            return $this->getModalContent(request()->get('entity'), 'create', $this->crud->getCreateFields());
+        }
+    }
+
+    public function setupOperationSettings()
+    {
+        if (method_exists($this, 'setupCreateOperation')) {
+            $this->setupCreateOperation();
+        } else {
+            $this->setup();
+        }
+
+        $this->crud->applyConfigurationFromSettings('create');
+    }
+
+    public function getModalContent($entity, $action, $fields)
+    {
+        return view(
+                'crud::fields.relationship.modal',
+                [
+                    'fields' => $fields,
+                    'action' => $action,
+                    'crud' => $this->crud,
+                    'entity' => $entity,
+                ]
+                );
+    }
+
+    public function InlineRefreshOptions()
+    {
+        $this->setupOperationSettings();
```

Comment:
One Hundred Percent Agreed :100: 

The problem I had that was pushing me back from do it the way you say is that I was not calling setup() before setupXxxOperation in case setupXxxOperation is present. I fixed it and now it works and follow convention. Thanks again :+1: